src/auth.js

```diff
@@ -4,32 +4,29 @@ const cognitoState = 'cognito_state';
 const cognitoUser = 'cognito_user';
 const cognitoAccessToken = 'cognito_access_token';
 
-// example: #access_token=header.eyJuYW1lIjoiSm9obiBEb2UifQ.signature&token_type=Bearer&state=1234
-const isResponse = /^#[\w]+=[\w.=-]+(&[\w]+=[\w.=-]+)*$/;
-
-// example: header.eyJuYW1lIjoiSm9obiBEb2UifQ.signature
-const isToken = /^[\w=-]+.[\w=-]+.[\w=-]+$/;
-
 let _AWSCredentials = null;
 
 export default class Auth {
 
   static initialize() {
-    // try parse and store the cognito response
-    // and fail silently otherwise
-    try {
-      const response = this.parseResponse(window.location.hash);
-      if (response.token_type === 'Bearer' && response.state === this.state()) {
-        this.setUser(this.parseToken(response.access_token));
-        this.setAccessToken(response.id_token);
+    if (window.location.hash.startsWith('#')) {
+      // https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html
+      const response = window.location.hash.substring(1);
+      const params = new URLSearchParams(response);
+      if (params.has('access_token') && params.has('id_token') &&
+          params.get('token_type') === 'Bearer' && params.get('state') === this.state()) {
+        localStorage.setItem('rawCognitoResponse', response);
```

Comment:
minor but `localStorage.removeItem('rawCognitoResponse');` could be added in `logout`

Reply:
but the item is never read, remove this line ?

Reply:
It is for debugging. The login process is very complicate and envolve opening and closing browser windows.
As a consequence it is not easy to see the cognito response (breakpoints do not work) when debugging, except by storing it in the local storage.